test/client/fixtures/sandbox/xhr-test.js

```diff
@@ -239,3 +241,49 @@ asyncTest('"XHR_COMPLETED_EVENT" should be raised when xhr is prevented (GH-1283
     xhr.open('GET', '/xhr-test/', true);
     xhr.send();
 });
+
+test('xhr request should emulate native behavior with error and error status code (GH-1397)', function () {
```

Comment:
should emulate a native browser behavior for xhr request with non success status code